File: kuji.py
import random
import os
from datetime import datetime

RECORD_FILE = 'kuji_record.txt'

def get_time_slot():
    now = datetime.now()
    date_str = now.strftime('%Y-%m-%d')
    if now.hour < 12:
        slot = 'AM'
    else:
        slot = 'PM'
    return f'{date_str}_{slot}'

def has_drawn_today():
    slot = get_time_slot()
    if os.path.exists(RECORD_FILE):
        with open(RECORD_FILE, 'r') as f:
            last_slot = f.read().strip()
            if last_slot == slot:
                return True
    return False

def record_draw():
    slot = get_time_slot()
    with open(RECORD_FILE, 'w') as f:
        f.write(slot)

def main():
    print("=== くじ引きプログラム（12時でリセット） ===")
    if has_drawn_today():
        print("今日はもうくじを引いています。12時を過ぎたらまた引けます。")
        return

    total = int(input("くじの本数を入力してください: "))
    win = int(input("当たりの本数を入力してください: "))
    
    if win > total or win < 0 or total <= 0:
        print("入力値が不正です。")
        return

    kujis = ['当たり'] * win + ['ハズレ'] * (total - win)
    random.shuffle(kujis)

    input("Enterキーでくじを引く...")
    result = kujis.pop()
    print(f"結果: {result}")
    record_draw()
    print("くじの記録を保存しました。12時を過ぎたらまた引けます。")

if __name__ == "__main__":
    main() 

File: kuji.js
const fs = require('fs');
const readline = require('readline');
const path = require('path');

const RECORD_FILE = path.join(__dirname, 'kuji_record.txt');

function getTimeSlot() {
  const now = new Date();
  const dateStr = now.toISOString().slice(0, 10); // YYYY-MM-DD
  const slot = now.getHours() < 12 ? 'AM' : 'PM';
  return `${dateStr}_${slot}`;
}

function hasDrawnToday() {
  const slot = getTimeSlot();
  if (fs.existsSync(RECORD_FILE)) {
    const lastSlot = fs.readFileSync(RECORD_FILE, 'utf8').trim();
    if (lastSlot === slot) {
      return true;
    }
  }
  return false;
}

function recordDraw() {
  const slot = getTimeSlot();
  fs.writeFileSync(RECORD_FILE, slot);
}

function askQuestion(query) {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
  });
  return new Promise(resolve => rl.question(query, ans => {
    rl.close();
    resolve(ans);
  }));
}

async function main() {
  console.log('=== くじ引きプログラム（12時でリセット） ===');
  if (hasDrawnToday()) {
    console.log('今日はもうくじを引いています。12時を過ぎたらまた引けます。');
    return;
  }

  const total = parseInt(await askQuestion('くじの本数を入力してください: '), 10);
  const win = parseInt(await askQuestion('当たりの本数を入力してください: '), 10);

  if (isNaN(total) || isNaN(win) || win > total || win < 0 || total <= 0) {
    console.log('入力値が不正です。');
    return;
  }

  const kujis = Array(win).fill('当たり').concat(Array(total - win).fill('ハズレ'));
  // シャッフル
  for (let i = kujis.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [kujis[i], kujis[j]] = [kujis[j], kujis[i]];
  }

  await askQuestion('Enterキーでくじを引く...');
  const result = kujis.pop();
  console.log(`結果: ${result}`);
  recordDraw();
  console.log('くじの記録を保存しました。12時を過ぎたらまた引けます。');
}

main(); 